### PR TITLE
Get Energia Compatibility

### PR DESCRIPTION
**Pull request by battosai30**
Hi,

Simple modifications that allow to get SSD1306 working on Energia ;)

### CODE REVIEW

File: Adafruit_SSD1306.h [hunk 1]
```diff
@@ -24,7 +24,7 @@ All text above, and the splash screen must be included in any redistribution
   #define WIRE_WRITE Wire.send
 #endif
 
-#ifdef __SAM3X8E__
+#if !defined(__SAM3X8E__) &&  !defined(ENERGIA)
```
Comment (fingolfin) [inline Adafruit_SSD1306.h]: That part of this PR seems wrong -- it changes the behavior on __SAM3X8E__ (it changes from "if __SAM3X8E__ defined" to "if __SAM3X8E__ NOT defined"

**Comment by fingolfin** [comment]
It look as if in several places, you changed the logic from 
```
#if defined(__SAM3X8E__)
```
to
```
#if !defined(__SAM3X8E__) &&  !defined(ENERGIA)
```
which seems weird -- note the negation for the `__SAM3X8E__` check. Seems wrong, but perhaps there is an explanation?

In any case, this PR does not even apply anymore. Parts of it may be obsolete, as the code was changed to check the TWBR directly. So I'd hope a much smaller PR might now suffice -- but I cannot test this, as I can't run the energia app on my machine (Mac OS X 10.11 -- Energia wants the Java 6 SDK, which I won't install for it).

**Comment by battosai30** [comment]
You're right, I don't know what I did but the first correction is good as it's really an #ifndef but the others are not. 

In my case it's was not a problem as this errors only affect arduino DUE and I was on Energia.

I'm not a big expert in Github, is it possible to correct this request or I have to create a new one ?

**Comment by microbuilder** [comment]
Fixed in #44

**Pull request closed without merging**